### PR TITLE
fix harness config && add automatic check(stable)

### PR DESCRIPTION
**Pull request by grybd**

### CODE REVIEW

File: .github/workflows/check.py [hunk 1]
```diff
@@ -377,11 +407,35 @@ def verify_model_entrys_in_readme(rows, rows_in_readme):
...
 
+
+def config_harness_name_should_be_standard(path):
+    for fn in walk(path):
+        if not fn.endswith('config.yaml'):
+            continue
+        with open(fn, mode="r", encoding="utf-8") as f:
+            config = yaml.load(f.read(), Loader=yaml.FullLoader)
+            if config.get('BM1684X') != None and config['BM1684X'].get('harness') != None:
+                harness_args_dict = config['BM1684X'].get('harness')
```
Comment (Jeffwhen) [inline .github/workflows/check.py:424]:
```
for harness in recursive_iter(config):
    assert xxx
```

**Review by Jeffwhen — CHANGES_REQUESTED** [review]

**Comment by Jeffwhen** [comment]
```python3
def key_value_iter(d):
    if type(d) == list:
        for v in d:
            for k, v in key_value_iter(v):
                yield k, v
    elif type(d) == dict:
        for dk, dv in d.items():
            for k, v in key_value_iter(dv):
                yield k, v
            yield dk, dv

def config_harness_name_should_be_standard(path):
    for fn in walk(path):
        if not fn.endswith('config.yaml'):
            continue
        with open(fn, mode='r', encoding='utf-8') as f:
            config = yaml.load(f.read(), Loader=yaml.FullLoader)
            name_set = set()
            for k, v in key_value_iter(config):
                if k != 'harness':
                    continue
                assert 'args' in v, f'harness.args does not exist in {fn}'
                for arg in v['args']:
                    assert 'name' in arg, f'harness.args.name does not exist in {fn}'
                name = arg['name']
                assert name not in name_set, \
                    f'Duplicate harness arg name \"{name}\" in {fn}'
                name_set.add(name)
```

**Comment by grybd** [comment]
> ```python
> def key_value_iter(d):
>     if type(d) == list:
>         for v in d:
>             for k, v in key_value_iter(v):
>                 yield k, v
>     elif type(d) == dict:
>         for dk, dv in d.items():
>             for k, v in key_value_iter(dv):
>                 yield k, v
>             yield dk, dv
> 
> def config_harness_name_should_be_standard(path):
>     for fn in walk(path):
>         if not fn.endswith('config.yaml'):
>             continue
>         with open(fn, mode='r', encoding='utf-8') as f:
>             config = yaml.load(f.read(), Loader=yaml.FullLoader)
>             name_set = set()
>             for k, v in key_value_iter(config):
>                 if k != 'harness':
>                     continue
>                 assert 'args' in v, f'harness.args does not exist in {fn}'
>                 for arg in v['args']:
>                     assert 'name' in arg, f'harness.args.name does not exist in {fn}'
>                 name = arg['name']
>                 assert name not in name_set, \
>                     f'Duplicate harness arg name \"{name}\" in {fn}'
>                 name_set.add(name)
> ```
已经修改，在这段代码基础上加上一个target_list 避免 不同target 的harness name 相同 出现误报的问题 @Jeffwhen

File: .github/workflows/check.py [hunk 2]
```diff
@@ -377,11 +407,42 @@ def verify_model_entrys_in_readme(rows, rows_in_readme):
...
+        with open(fn, mode='r', encoding='utf-8') as f:
+            config = yaml.load(f.read(), Loader=yaml.FullLoader)
+            for k, v in key_value_iter(config):
+                name_set = set()
+                if k != 'harness':
+                    continue
+                assert 'args' in v, f'harness.args does not exist in {fn}'
+                for arg in v['args']:
+                    assert 'name' in arg, f'harness.args.name does not exist in {fn}'
+                name = arg['name']
```
Comment (Jeffwhen) [inline .github/workflows/check.py:440]: 这几行缩进不对

File: .github/workflows/check.py [hunk 3]
```diff
@@ -377,11 +407,42 @@ def verify_model_entrys_in_readme(rows, rows_in_readme):
...
+            yield dk, dv
+
+def config_harness_name_should_be_standard(path):
+    for fn in walk(path):
+        if not fn.endswith('config.yaml'):
+            continue
+        with open(fn, mode='r', encoding='utf-8') as f:
+            config = yaml.load(f.read(), Loader=yaml.FullLoader)
+            for k, v in key_value_iter(config):
+                name_set = set()
```
Comment (Jeffwhen) [inline .github/workflows/check.py:434]: 这一行直接诺到 438 行。

**Review by Jeffwhen — CHANGES_REQUESTED** [review]

File: .github/workflows/check.py (hunk 3 above)
Comment (grybd) [reply .github/workflows/check.py]: 已修改

File: .github/workflows/check.py (hunk 2 above)
Comment (grybd) [reply .github/workflows/check.py]: 已修改

**Pull request merged**